packages/application-extension/src/index.ts

```diff
@@ -556,6 +597,158 @@ const topVisibility: JupyterFrontEndPlugin<void> = {
+            'Show %1 in the left sidebar',
+            args['title'] as string
+          );
+        } else if (args['side'] === 'right') {
+          return trans.__(
+            'Show %1 in the right sidebar',
+            args['title'] as string
+          );
+        }
+        return trans.__('Show %1 in the sidebar', args['title'] as string);
+      },
+      execute: args => {
+        switch (args['side'] as string) {
+          case 'left':
+            if (notebookShell.leftCollapsed) {
+              notebookShell.activateById(args['id'] as string);
+            } else {
+              notebookShell.collapseLeft();
+              if (notebookShell.currentWidget) {
+                notebookShell.activateById(notebookShell.currentWidget.id);
+              }
+            }
+            break;
+          case 'right':
+            if (notebookShell.rightCollapsed) {
+              notebookShell.activateById(args['id'] as string);
+            } else {
+              notebookShell.collapseRight();
+              if (notebookShell.currentWidget) {
+                notebookShell.activateById(notebookShell.currentWidget.id);
+              }
+            }
+            break;
+        }
+      },
+      isToggled: args => {
+        if (notebookShell.leftCollapsed) {
+          return false;
+        }
+        const currentWidget = notebookShell.leftHandler.current;
+        if (!currentWidget) {
+          return false;
+        }
+
+        return currentWidget.id === (args['id'] as string);
+      }
+    });
+
+    /**
+     * The function which adds entries to the View menu for each widget of a sidebar.
+     *
+     * @param area - 'left' or 'right', the area of the side bar.
+     * @param entryLabel - the name of the main entry in the View menu for that sidebar.
+     * @returns - The disposable menu added to the View menu or null.
+     */
+    const updateMenu: SideBarPanel.UpdateSideBarMenuFn = (area, entryLabel) => {
+      if (menu === null) {
+        return null;
+      }
+
+      let disposableMenu: IDisposable | null = null;
+
+      const newMenu = new Menu({ commands: app.commands });
+      newMenu.title.label = entryLabel;
+      const widgets = notebookShell.widgets(area);
+      let menuToAdd = false;
+
+      each(widgets, widget => {
+        newMenu.addItem({
+          command: CommandIDs.togglePanel,
+          args: {
+            side: area,
+            title: `Show ${widget.title.caption}`,
+            id: widget.id
+          }
+        });
+        menuToAdd = true;
+      });
+
+      // If there are widgets, add the menu to the main menu entry.
+      if (menuToAdd) {
+        disposableMenu = menu.viewMenu.addItem({
+          type: 'submenu',
+          submenu: newMenu
+        });
+      }
+
+      return disposableMenu;
+    };
+
+    app.restored.then(() => {
+      // Add the notebook tools in right area.
+      if (notebookTools) {
+        notebookShell.add(notebookTools, 'right');
```

Comment:
It looks like this shouldn't be in the `@jupyter-notebook/notebook-extension` extension rather than `@jupyter-notebook/application-extension` if we want to include it by default?

Reply:
I don't understand what you mean. What is in the `@jupyter-notebook/notebook-extension` ?

Reply:
Here are the plugins more related to the notebook than general: https://github.com/jupyter/notebook/blob/main/packages/notebook-extension/src/index.ts

Reply:
OK, the "shouldn't" misled me.
https://github.com/jupyter/notebook/pull/6487#pullrequestreview-1081591262 is useless as it was the same idea but in `@jupyter-notebook/application-extension`

Reply:
Ah sorry for the typo, I corrected it in the message above.